main/src/library/MutList.flix

```diff
@@ -29,10 +29,11 @@ instance ToString[MutList[a]] with ToString[a] {
 }
 
 instance Eq[MutList[a]] with Eq[a] {
-    pub def eq(xs: MutList[a], ys: MutList[a]): Bool =
-        (let MutList(a, _) = xs;
-        let MutList(b, _) = ys;
-        deref a == deref b) as & Pure
+        pub def eq(xs: MutList[a], ys: MutList[a]): Bool = {
+            let MutList(a, k) = xs;
+            let MutList(b, l) = ys;
+            deref a == deref b and deref k == deref l
```

Comment:
I don't think this is correct.

What matters is if they have the same elements, not if their arrays are the same with the same capacities.

I think the arrays should be sliced up to length l and k, and then compared.